### PR TITLE
Fix missing links for registered app with device

### PR DESCRIPTION
**Pull request by VVeremjova**
Fixed defect [APPLINK-30858](https://adc.luxoft.com/jira/browse/APPLINK-30858)

### CODE REVIEW

**Comment by VVeremjova** [comment]
@dev-gh, @dtrunov  @AByzhynar
Please review

File: src/components/application_manager/src/policies/policy_handler.cc [hunk 1]
```diff
@@ -678,6 +668,25 @@ void PolicyHandler::OnGetListOfPermissions(const uint32_t connection_key,
   }
 }
 
+void PolicyHandler::LinkAppsToDevice() {
+  sync_primitives::AutoLock lock(app_to_device_link_lock_);
+  LinkAppToDevice linker(app_to_device_link_, application_manager_);
+  LOG4CXX_DEBUG(logger_, "add links to app.  no specific app was passed");
+  {
+    const ApplicationSet& accessor =
```
Comment (AByzhynar) [inline src/components/application_manager/src/policies/policy_handler.cc]: @VVeremjova 
```
if (accessor.empty()) {
 // warn
} else {
std::for_each(accessor.begin(), accessor.end(), linker);
}
```

Comment (VVeremjova) [inline src/components/application_manager/src/policies/policy_handler.cc]: @AByzhynar See b04da10

**Comment by AByzhynar** [comment]
@VVeremjova Please don't forget to squash commits

**Pull request merged**